File: FangZhiHu/app/src/main/java/com/tongqinyuan/fangzhihu/OnRequestListener.java
package com.tongqinyuan.fangzhihu;


import qj.ccmtjf.com.lib.api.ApiResponse;

/**
 * Created by tongqinyuan on 2017/11/4.
 */

public interface OnRequestListener<T> {

    /**
     * @param response 返回调用传入的泛型对象
     */
    void onSuccess(T response);

    void onFailure(ApiResponse<String> errorApiResponse);
}

File: FangZhiHu/CommonLibKotlin/src/main/java/com/ytq/main/common/OnRequestListener.kt
package com.ytq.main.common

import network.entity.ApiResponse

/**
 * Created by tongqinyuan on 2018/3/9.
 */
interface OnRequestListener<T> {

    /**
     * @param response 返回调用传入的泛型对象
     */
     fun onSuccess(response: T)

     fun onFailure(errorApiResponse: ApiResponse<String>)

}